go/pkg/schema/languages.go

```diff
@@ -0,0 +1,5132 @@
+package schema
+
+type Languages struct {
```

Comment:
Wow this is big struct.
Would it be a problem if it is a map of structs?
This will also avoid a situation where you will have to remember add new languages to both`languages.yaml` and to `languages.go` 

```go

type LanguageProprties struct {
	Type               string   `yaml:"type,omitempty"`
	Color              string   `yaml:"color,omitempty"`
	Extensions         []string `yaml:"extensions,omitempty"`
	TmScope            string   `yaml:"tm_scope,omitempty"`
	AceMode            string   `yaml:"ace_mode,omitempty"`
	LanguageID         int      `yaml:"language_id,omitempty"`
	Aliases            []string `yaml:"aliases,omitempty"`
	CodemirrorMode     string   `yaml:"codemirror_mode,omitempty"`
	CodemirrorMimeType string   `yaml:"codemirror_mime_type,omitempty"`
}

type Languages map[string]LanguageProprties

```

Reply:
Only downside here is that `LanguageProprties` has to be same for all languages, but that should be ok, as the fields that are not populated in yaml file will be empty.

Reply:
Yes, that was the first thing i tried to make it work but i would end up to something similar to what you're suggesting eventually. So +1.  Golang reflection is really a pain for newbie :sweat_smile:

Reply:
Don't worry it does not get better. :rofl:  Golang's reflection is  pain for everyone :smile: